e2e_test.go

```diff
@@ -0,0 +1,1035 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/vdaas/vald-client-go/v1/agent/core"
+	"github.com/vdaas/vald-client-go/v1/payload"
+	"github.com/vdaas/vald-client-go/v1/vald"
+	"google.golang.org/grpc"
+)
+
+const addr = "localhost:8081"
+
+var (
+	conn *grpc.ClientConn
+
+	client      vald.Client
+	agentClient core.AgentClient
+
+	ctx context.Context
+
+	data []Datum
+
+	insertConfig = &payload.Insert_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	searchNum    = 3
+	searchConfig = &payload.Search_Config{
+		Num:     uint32(searchNum),
+		Radius:  -1.0,
+		Epsilon: 0.1,
+		Timeout: 3000000000,
+	}
+
+	updateConfig = &payload.Update_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	upsertConfig = &payload.Upsert_Config{
+		SkipStrictExistCheck: true,
+	}
+
+	removeConfig = &payload.Remove_Config{
+		SkipStrictExistCheck: true,
+	}
+)
+
+type Datum struct {
+	ID     string    `json:"id"`
+	Vector []float32 `json:"vector"`
+}
+
+func init() {
+	ctx = context.Background()
+
+	err := readJSON()
+	if err != nil {
+		fmt.Errorf("error: %s", err)
+		os.Exit(1)
+	}
+}
+
+func readJSON() error {
+	f, err := os.Open("wordvecs1000.json")
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	bytes, err := ioutil.ReadAll(f)
+	if err != nil {
+		return err
+	}
+
+	return json.Unmarshal(bytes, &data)
+}
+
+func beforeEach() {
+	var err error
+
+	conn, err = grpc.DialContext(ctx, addr, grpc.WithInsecure())
+	if err != nil {
+		fmt.Errorf("error: %s", err)
+		os.Exit(1)
+	}
+
+	client = vald.NewValdClient(conn)
+	agentClient = core.NewAgentClient(conn)
+}
+
+func afterEach() {
+	conn.Close()
+}
+
+func TestE2E(t *testing.T) {
+	// Insert
+
+	t.Run(`Test for Insert operation`, func(t *testing.T) {
+		beforeEach()
+		defer afterEach()
```

Comment:
Should we use `t.Cleanup ` instead of `defer` for teardown? 🤔 
If I made a mistake, I am sorry.

https://pkg.go.dev/testing#T.Cleanup

Reply:
I didn't know about it. Thanks!